src/helpers.py

```diff
@@ -0,0 +1,154 @@
+
+    def _cleanup_msgs(self):
+        while len(self.msgs) > self.MAX_MSGS_SIZE:
+            self.msgs.popitem(last=False)  # remove oldest item
+
+
+class CapturedData:
+    """
+    CapturedData is image data with the data as an np array,
+    plus the timestamp it was captured at.
+    """
+
+    def __init__(self, np_array: NDArray, captured_at: float) -> None:
+        self.np_array = np_array
+        self.captured_at = captured_at
+
+
+def encode_depth_raw(data: bytes, shape: Tuple[int, int], logger: Logger) -> bytes:
+    """
+    Encodes raw data into a bytes payload deserializable by the Viam SDK (camera mime type depth)
+
+    Args:
+        data (bytes): raw bytes
+        shape (Tuple[int, int]): output dimensions of depth map (height x width)
+
+    Returns:
+        bytes: encoded bytes
+    """
+    height, width = shape  # using np shape for actual output's height/width
+    MAGIC_NUMBER = struct.pack(
+        ">Q", 4919426490892632400
+    )  # UTF-8 encoding for 'DEPTHMAP'
+    MAGIC_BYTE_COUNT = struct.calcsize("Q")
+    WIDTH_BYTE_COUNT = struct.calcsize("Q")
+    HEIGHT_BYTE_COUNT = struct.calcsize("Q")
+
+    # Depth header contains 8 bytes for the magic number, followed by 8 bytes for width and 8 bytes for height. Each pixel has 2 bytes.
+    pixel_byte_count = np.dtype(np.uint16).itemsize * width * height
+    width_to_encode = struct.pack(">Q", width)  # Q signifies big-endian
+    height_to_encode = struct.pack(">Q", height)
+    total_byte_count = (
+        MAGIC_BYTE_COUNT + WIDTH_BYTE_COUNT + HEIGHT_BYTE_COUNT + pixel_byte_count
+    )
+    logger.debug(
+        f"Calculated size:  {MAGIC_BYTE_COUNT} + {WIDTH_BYTE_COUNT} + {HEIGHT_BYTE_COUNT} + {pixel_byte_count} = {total_byte_count}"
+    )
+    logger.debug(f"Actual data size: {len(data)}")
+
+    # Create a bytearray to store the encoded data
+    raw_buf = bytearray(total_byte_count)
+    offset = 0
+
+    # Copy the depth magic number into the buffer
+    raw_buf[offset : offset + MAGIC_BYTE_COUNT] = MAGIC_NUMBER
+    offset += MAGIC_BYTE_COUNT
+
+    # Copy the encoded width and height into the buffer
+    raw_buf[offset : offset + WIDTH_BYTE_COUNT] = width_to_encode
+    offset += WIDTH_BYTE_COUNT
+    raw_buf[offset : offset + HEIGHT_BYTE_COUNT] = height_to_encode
+    offset += HEIGHT_BYTE_COUNT
+
+    # Copy data into rest of the buffer
+    raw_buf[offset : offset + pixel_byte_count] = data
+    return bytes(raw_buf)
+
+
+def encode_jpeg_bytes(arr: NDArray, is_depth: bool = False) -> bytes:
+    """
+    Encodes numpy color image data into a bytes decodable in the JPEG image format.
+
+    Args:
+        arr (NDArray): frame
+        is_depth (bool): whether arr is depth data or RGB data
+
+    Returns:
+        bytes: JPEG bytes
+    """
+    if is_depth:
+        pil_image = Image.fromarray(arr, "I;16").convert("RGB")
+    else:
+        pil_image = Image.fromarray(arr)
+    output_buffer = io.BytesIO()
```

Comment:
[possible nit] Should we use `with` here to to ensure resource is closed out.

```
        with io.BytesIO() as output_buffer:
            pil_image.save(output_buffer, format="JPEG")
            raw_bytes = output_buffer.getvalue()

Reply:
Done